Gems/ROS2/Code/Source/Camera/PostProcessing/ROS2ImageEncodingConvertComponent.cpp

```diff
@@ -0,0 +1,185 @@
+
+            // Perform conversion in place
+            for (size_t pixelId = 0; pixelId < image.width * image.height; ++pixelId)
+            {
+                size_t pixelOffsetIn = pixelId * 4;
+                size_t pixelOffsetOut = pixelId * 3;
+                image.data[pixelOffsetOut] = image.data[pixelOffsetIn];
+                image.data[pixelOffsetOut + 1] = image.data[pixelOffsetIn + 1];
+                image.data[pixelOffsetOut + 2] = image.data[pixelOffsetIn + 2];
+            }
+            image.encoding = "rgb8";
+            image.step = image.width * 3;
+            image.data.resize(image.step * image.height);
+        }
+
+        const AZStd::unordered_map<ImageEncoding, const char*> ImageEncodingNames = {
+            { ImageEncoding::rgba8, "rgba8" },
+            { ImageEncoding::rgb8, "rgb8" },
+            { ImageEncoding::mono8, "mono8" },
+            { ImageEncoding::mono16, "mono16" },
+        };
+        const AZStd::unordered_map<AZStd::string, ImageEncoding> ImageEncodingFromName = {
+            { "rgba8", ImageEncoding::rgba8 },
+            { "rgb8", ImageEncoding::rgb8 },
+            { "mono8", ImageEncoding::mono8 },
+            { "mono16", ImageEncoding::mono16 },
+        };
+
+        const AZStd::unordered_map<EncodingConverion, const AZStd::function<void(sensor_msgs::msg::Image&)>> supportedFormatChange = {
+            { { ImageEncoding::rgba8, ImageEncoding::rgb8 }, Rgba8ToRgb8 },
+        };
+    } // namespace
+
+    void EncodingConverion::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<EncodingConverion>()
+                ->Version(0)
+                ->Field("EncodingIn", &EncodingConverion::encodingIn)
+                ->Field("EncodingOut", &EncodingConverion::encodingOut);
+
+            if (AZ::EditContext* ec = serializeContext->GetEditContext())
+            {
+                ec->Class<EncodingConverion>("Encoding Conversion", "Specifies encoding conversion")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::ComboBox, &EncodingConverion::encodingIn, "Encoding In", "Encoding of the input image")
+                    ->EnumAttribute(ImageEncoding::rgba8, "rgba8")
+                    ->EnumAttribute(ImageEncoding::rgb8, "rgb8")
+                    ->EnumAttribute(ImageEncoding::mono8, "mono8")
+                    ->EnumAttribute(ImageEncoding::mono16, "mono16")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::ComboBox, &EncodingConverion::encodingOut, "Encoding Out", "Encoding of the output image")
+                    ->EnumAttribute(ImageEncoding::rgba8, "rgba8")
+                    ->EnumAttribute(ImageEncoding::rgb8, "rgb8")
+                    ->EnumAttribute(ImageEncoding::mono8, "mono8")
+                    ->EnumAttribute(ImageEncoding::mono16, "mono16");
```

Comment:
Don't you want to validate at this point if the input format is different from the output format? You could do it in `ROS2ImageEncodingConvertComponent::ValidateEncodingConversion` as well.